packages/manager/src/features/Linodes/LinodeCreate/Summary/Summary.tsx

```diff
@@ -28,6 +33,34 @@ export const Summary = ({ isAlertsBetaMode }: SummaryProps) => {
 
   const { control } = useFormContext<LinodeCreateFormValues>();
 
+  const [stackscriptData] = useWatch({
+    control,
+    name: ['stackscript_data'],
+  });
+
+  const rawClusterData = parseClusterData(stackscriptData);
+
+  const { data: allTypes } = useAllTypes();
+
+  const labelToIdMap = React.useMemo(() => {
+    if (!allTypes) return {};
+    return allTypes.reduce<Record<string, string>>((acc, type) => {
+      acc[type.label] = type.id;
+      return acc;
+    }, {});
+  }, [allTypes]);
+
+  const typeIds = rawClusterData
+    .map((c) => labelToIdMap[c.typeId ?? ''])
+    .filter((id): id is string => Boolean(id));
+
+  const clusterTypeQueries = useSpecificTypes(typeIds);
+
+  const clusterData = rawClusterData.map((cluster, i) => ({
+    ...cluster,
+    typeData: clusterTypeQueries[i]?.data,
+  }));
```

Comment:
I think we may be able to simplify this:

```suggestion
  const { data: allTypes } = useAllTypes();

  const rawClusterData = parseClusterData(stackscriptData);

  const clusterData = rawClusterData.map((cluster) => ({
    ...cluster,
    typeData: allTypes?.find((t) => t.label === cluster.typeId),
  }));
```

Rather than having two `useWatch` calls in the component, we can add `stackscript_data` to the existing `useWatch` call and remove the `useWatch` added by this PR